Lib/test/test_compile.py

```diff
@@ -1385,52 +1385,90 @@ def check_op_count(func, op, expected):
             self.assertEqual(actual, expected)
 
         def check_consts(func, typ, expected):
-            slice_consts = 0
+            all_consts = set()
```

Comment:
I don't think this is working correctly - slices just defer to their contents for hash/equality, so this will be deduplicating the slices in the same way this is trying to fix.

Reply:
Ooh, nice catch!